RadixWallet/Clients/ProfileStore/ProfileStore.swift

```diff
@@ -288,364 +223,306 @@ extension ProfileStore {
+		@Dependency(\.secureStorageClient) var secureStorageClient
+		var headers = try secureStorageClient.loadProfileHeaderList()?.rawValue ?? []
+		headers[id: header.id] = header
+		if let headerList = NonEmpty(rawValue: headers) {
+			try secureStorageClient.saveProfileHeaderList(headerList)
+		} else {
+			struct FailedToUpdateHeaderListWasEmpty: Swift.Error {}
+			throw FailedToUpdateHeaderListWasEmpty()
+		}
 	}
 
-	private func lens<Property>(
-		_ keyPath: KeyPath<ProfileState, Property>
-	) -> AnyAsyncSequence<Property> where Property: Sendable & Equatable {
-		lens { $0[keyPath: keyPath] }
+	private static func _setActiveProfile(to header: ProfileSnapshot.Header) {
+		@Dependency(\.userDefaultsClient) var userDefaultsClient
+		userDefaultsClient.setActiveProfileID(header.id)
 	}
 
-	private func lens<Property>(
-		_ transform: @escaping @Sendable (ProfileState) -> Property?
-	) -> AnyAsyncSequence<Property> where Property: Sendable & Equatable {
-		profileStateSubject.compactMap(transform)
-			.share() // Multicast
-			.removeDuplicates()
-			.eraseToAnyAsyncSequence()
+	private static func _persist(bdfsMnemonic: PrivateHDFactorSource) throws {
+		@Dependency(\.secureStorageClient) var secureStorageClient
+		try secureStorageClient.saveMnemonicForFactorSource(bdfsMnemonic)
 	}
 
-	private func changeState(to newState: ProfileState) {
-		profileStateSubject.send(newState)
+	private static func _persist(profile: Profile) throws {
+		try _persist(profileSnapshot: profile.snapshot())
 	}
-}
 
-extension ProfileStore {
-	static func createDeviceInfo() async throws -> ProfileSnapshot.Header.UsedDeviceInfo {
-		@Dependency(\.date) var dateGenerator
-		@Dependency(\.device) var device
-		@Dependency(\.uuid) var uuid
+	private static func _persist(profileSnapshot: ProfileSnapshot) throws {
 		@Dependency(\.secureStorageClient) var secureStorageClient
-
-		let date = dateGenerator.now
-
-		let deviceIdentifier: UUID = if let existing = try? await secureStorageClient.loadDeviceIdentifier() {
-			existing
-		} else {
-			uuid()
-		}
-
-		let description = await NonEmptyString(rawValue: "\(device.name) (\(device.model))")!
-
-		return ProfileSnapshot.Header.UsedDeviceInfo(
-			description: description,
-			id: deviceIdentifier,
-			date: date
-		)
+		try secureStorageClient.saveProfileSnapshot(profileSnapshot)
 	}
 }
 
-// MARK: - EphemeralProfile
-struct EphemeralProfile: Sendable, Hashable {
-	var profile: Profile
-	/// If this during startup an earlier Profile was found but we failed to load it.
-	let loadFailure: Profile.LoadingFailure?
+// MARK: - MetaDeviceInfo
+private struct MetaDeviceInfo: Sendable, Hashable {
+	let deviceInfo: DeviceInfo
+	let fromDeprecatedDeviceID: Bool
 }
 
-extension UserDefaultsClient {
-	public func getActiveProfileID() -> ProfileSnapshot.Header.ID? {
-		stringForKey(.activeProfileID).flatMap(UUID.init(uuidString:))
+extension OverlayWindowClient.Item.AlertState {
+	public static func profileUsedOnAnotherDeviceAlert(
```

Comment:
Not quite sure about this alert, should probably be a proper sheet that can display a better info. But this is a product decision.

Reply:
Mmm lets leave it to Matt